script.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-unused-vars */
+const bookContainer = document.querySelector('.books-container');
+const bookTitle = document.querySelector('#title');
+const bookAuthor = document.querySelector('#author');
+const addBtn = document.querySelector('.add-btn');
+
+let bookList = [];
+
+function bookUpload() {
+  bookContainer.innerHTML = '';
+  for (let i = 0; i < bookList.length; i += 1) {
+    bookContainer.innerHTML += `
+    <div class="books">
+    <h3>${bookList[i].bookTitle}</h3>
+    <h3>${bookList[i].bookAuthor}</h3>
+    <button onclick="deleted(${i})">Remove</button>
+    <hr />
+  </div>
+   `;
+    bookTitle.value = '';
+    bookAuthor.value = '';
+  }
+}
+
+function deleted(index) {
+  bookList.splice(index, 1);
+  bookUpload();
+  localStorage.setItem('bookList', JSON.stringify(bookList));
+}
+
+window.onload = () => {
+  if (localStorage.getItem('bookList')) {
+    bookList = JSON.parse(localStorage.getItem('bookList'));
+  }
+  bookUpload();
+};
+
+addBtn.addEventListener('click', () => {
+  const books = {
+    bookTitle: bookTitle.value,
+    bookAuthor: bookAuthor.value,
+  };
+  bookList.push(books);
+  bookUpload();
+  localStorage.setItem('bookList', JSON.stringify(bookList));
```

Comment:
- [ ] Kindly make a function for adding book and then call that function in this eventlistener to make your function reusable 😊

Reply:
@HAMMAS-SALEEM  can you please explain better on creating a function, is that supposed to be optional or a must

Reply:
@shegy28 
As you can see that you have a function for removing a book and for rendering the books on your dom. Similar to that you should have a function for creating a book which you can call in the eventListeners :blush:

```javascript
function addBooks() {
const books = {
    bookTitle: bookTitle.value,
    bookAuthor: bookAuthor.value,
  };
  bookList.push(books);
  bookUpload();
  localStorage.setItem('bookList', JSON.stringify(bookList));
  }
  
  addBtn.addEventListener('click', () => {
  addBook();
});
  
```

Reply:
Oh thanks done that already